client/llb/fileop_test.go

```diff
@@ -178,6 +178,41 @@ func TestFileMkfile(t *testing.T) {
 	require.Equal(t, int64(-1), mkdir.Timestamp)
 }
 
+func TestFileSymlink(t *testing.T) {
+	t.Parallel()
+
+	st := Scratch().
+		File(Mkfile("/foo", 0700, []byte("data"))).
+		File(Symlink("foo", "/bar"))
```

Comment:
Can we have some tests that chain symlink together with other actions. Atm it is unclear what the point of `Mkfile` is in this test as it is in a different `FileOp` that is never checked after marshaling.

Reply:
Probably the result of copying and pasting from another test. To be clear about your intent, you want to be sure that `llb.Symlink` plays nicely with other file actions when chained together, i.e. they happen in the right order?

Reply:
Correct. The actions are all linked together so that the input points to either the input of the fileop or the offset of another action that needs to run before it

Reply:
To clarify, what do you mean by "chain"? Do you want to see a single `Op` with multiple actions? Or do you want "chained" calls to `.File(...)`?

In other words, do you want the following?
```go
state.File(
    MkDir(...).
    MkFile(...).
    MkDir(...).
    Symlink(...),
)
```

Or are you looking for something like this?
```go
state.File(MkDir(...)).
    File(MkFile(...)).
    File(MkDir(...)).
    File(Symlink(...))
```

Sorry for the holidelay